frankenstein/ring.go

```diff
@@ -65,6 +65,18 @@ func (r *Ring) Get(key uint64) (Collector, error) {
 	return r.circle[r.sortedHashes[i]], nil
 }
 
+// GetAll returns all collectors in the circle.
+func (r *Ring) GetAll() []Collector {
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+
+	collectors := make([]Collector, 0, len(r.circle))
+	for _, c := range r.circle {
+		collectors = append(collectors, c)
+	}
+	return collectors
+}
```

Comment:
Changed this and then also removed the set of already queried hosts in the caller, since I now assume I get every host only once due to IDs being hostnames. I haven't tested everything end to end again, so maybe have a look just in case :)